sprocs/group_get_displayed_role.sql

```diff
@@ -0,0 +1,18 @@
+CREATE FUNCTION
+    group_get_displayed_role (
+        IN user_ids bigint[],
+        IN course_instance_id bigint,
+        OUT roles text[]
+    )
+AS $$
+DECLARE 
+    user_id integer;
+BEGIN
+    IF user_ids IS NOT NULL THEN
+        FOREACH user_id IN ARRAY user_ids
+        LOOP
+            roles = array_append(roles, users_get_displayed_role(user_id, course_instance_id));
+        END LOOP;
+    END IF;
+END;
+$$ LANGUAGE plpgsql STABLE;
```

Comment:
I'm not convinced that a separate sproc is warranted in this case, instead of adding a column to the `group_info` sproc instead. This may require some consideration as to the performance impact if the sproc is used elsewhere that doesn't require roles, but there could be options to remediate that. Also, it would probably make sense to have this computation be done through a query (`array_agg`) instead of a loop, and to return it with unique roles.

Reply:
I agree with removing the need for a loop, we probably can rewrite the query from scratch and not rely on `users_get_displayed_roles`.

We can probably place the logic in the existing `group_info` sproc, although I'm not too sure how performance for other queries will be affected. Do you have any suggestions on how the performance impact can be minimized for use cases where roles aren't required?

Reply:
I just did a quick scan of all places that use this sproc, and most of them could be simplified by removing the sproc call altogether. In essence, these three references could easily replace `gi.id` with `ai.group_id` which would remove all need for this PR in general.
* `pages/instructorAssessmentStatistics/instructorAssessmentStatistics.sql`
* `sprocs/assessment_stats.sql`
* `sprocs/assessment_duration_stats.sql`

This leaves two places: this page and the API endpoint for assessment instances. I don't believe the API call would have an impact that I'd consider worrisome, so I think you can easily add the student role as a column without problems, and remove the reference to this sproc in the files above.

As for `users_get_displayed_role`, I believe it is better to keep its use for consistent computation of the role, but use it inside the query instead of in a loop.

And this issue is by no means urgent, if you can't work on it for a few days it's understandable.